pep-9999.rst

```diff
@@ -0,0 +1,247 @@
+
+    def cumulative_sums(data, start=0)
+        total = start
+        yield from (total := operator.iadd(total, value) for value in data)
+        return total
+
+The proposal in this PEP would allow the above code to work, while also allowing
+inline augmented assignment operations like the following::
+
+    remaining_attempts = 10
+    while remaining_attempts -= 1:
+        ... # Attempt the operation once
+
+With a preceding local name binding or declaration, it would allow name binding
+in if/elif chains::
+
+    m = pattern.match(data)
+    if m:
+        ...
+    elif m := other_pattern.match(data):
+        ...
+    elif m := yet_another_pattern.match(data):
+        ...
+    else:
+        ...
+
+    # Or equivalently
+    m : typing.Match
+    if m := pattern.match(data):
+        ...
+    elif m := other_pattern.match(data):
+        ...
+    elif m := yet_naother_pattern.match(data):
+        ...
+    else:
+        ...
+
+
+Note that this PEP also delegates the task of deciding whether or not it's a
+good idea to implicitly allow nonlocal name rebinding operations to PEP 572 -
+it's a competing PEP aimed at ensuring the resulting capability is at least
+moderately teachable (think "If we do this at all, we should do it this
+particular way, and explain it to end users with this particular semantic
+framing" rather than seeing it as an independent proposal), and the PEP author
+would be entirely happy with an outcome where both PEPs ended up being rejected
+(as if we decide "not right now", it's straightforward to reconsider inline
+assignment support at a later date, while once we accept and release it, we're
+going to be stuck with it indefinitely).
+
+
+Syntax and semantics
+====================
+
+Augmented assignment expressions
+--------------------------------
+
+The language grammar would be adjusted to allow augmented assignments to appear
+as expressions, where the result of the expression is the same reference that
+is being bound to the given target.
+
+When used as anything other than a standalone expression statement, the
+assignment target would be restricted to a simple name (see design discussion
+for more background on this).
+
+
+Augmented assignment target scoping
+-----------------------------------
+
+If augmented assignment expressions were added with their current scoping
+semantics (where they implicitly bind a local name, just like a regular
+assignment statement), then code like ``(total += value for value in data)``
+would fail at runtime with ``UnboundLocalError``, just as the following code
+does today::
+
+    def cumulative_sums(data, start=0)
+        total = start
+        def _inner_gen():
+            for value in data:
+                total += value
+                yield total
+        yield from _inner_gen()
+        return total
+
+To address this, the compilation process will be changed to handle augmented
+assignment target scope resolution as follows:
```

Comment:
After reading the rules, I found myself wishing I had examples to reference for each.  Mentioning the later examples here would help.  So would having at least a short example of each.

Reply:
Yeah, that's a good idea.